src/List.tsx

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from 'react';
+
+const List = ({ data, setData }: { data: any[]; setData: () => void; page: number }) => {
+  const [items, setItems] = useState<{ name: string }[]>([]);
+
+  useEffect(() => {
+    setItems(data.filter((i) => !i.isHidden));
+  }, []);
+
+  return (
+    <div>
+      {items.map((a) => (
+        <div>{a.name}</div>
+      ))}
```

Comment:
You can check for nonHidden items inside the map fn and avoid another state inside list component